File: pset1/cash/cash.c
#include <math.h>
#include <cs50.h>
#include <stdio.h>

int main(void)
{

//get number from user;
// n = cash number;
    float n;

    do
    {
        n = get_float("change owed: ");

    }
    while (n < 0);

//covert the dollars to cents;

    int cents = round(n * 100);
    printf("%i\n", cents);

// qn = quarter number, dn = dime number, nn = nickel number, pn = penny number;
  
    int qn = cents / 25;
    // printf("%i\n",qn);
      
    int dn = (cents % 25) / 10;
    //printf("%i\n",dn);
     
    int nn = (cents % 25 % 10) / 5;
    //printf("%i\n",nn);
    
    int pn = cents - qn * 25 - dn * 10 - nn * 5;
    //printf("%i\n",pn);
    
  
    //print out the number of the coins;
    
    int coins = qn + dn + nn + pn;
    printf("%i\n", coins);
}
  
//     int qn, dn, nn, pn, r, coins;
  
//     qn=0;
//     dn=0;
//     nn=0;
//     pn=0;
//   r=0;
 
 

//   qn=cents/25;
//   r = cents%25;
//   // printf("qn: %i\n", qn);
//   // printf("r: %i\n", r);
  
//   if(r > 0)
//   {
//     dn=r/10;
//     r=r%10;
//     // printf("dn: %i\n", dn);
//     // printf("r2: %i\n", r);
//   }
//   if(r > 0)
//   {
//     nn=r/5;
//     r=r%5;
//     // printf("nn: %i\n",nn);
//     // printf("r3: %i\n", r);
//   }
//   if(r > 0)
//   {
//     pn=r;
//     // printf("pn: %i\n",pn);
//     // printf("r4: %i\n", r);
//   }

//   coins = qn+dn+nn+pn;
//   printf("%i\n",coins);
// }

File: pset6/cash/cash.py
# from cs50 import get_float


# get number from user
# n = cash number
while True:
    try:
        # n = get_float("change owed: ")
        n = float(input("change owed: "))
        if n > 0:
            break
    except ValueError:
        print("That's not an number.")
        
# covert the dollars to cents
cents = round(n * 100)
# print(cents)

# qn = quarter number, dn = dime number, nn = nickel number, pn = penny number
qn = int(cents / 25)
# printf("%i\n",qn);

dn = int((cents % 25) / 10)
# printf("%i\n",dn);
     
nn = int((cents % 25 % 10) / 5)
# printf("%i\n",nn);
    
pn = int(cents - qn * 25 - dn * 10 - nn * 5)
# printf("%i\n",pn);

# print out the number of the coins;
coins = qn + dn + nn + pn
print(coins)